core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,190 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.iceberg.expressions.Literal;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.relocated.com.google.common.io.BaseEncoding;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.DateTimeUtil;
+
+public class DefaultValueParser {
+  private DefaultValueParser() {
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  public static Object parseDefaultFromJson(Type type, JsonNode defaultValue) {
+
+    if (defaultValue == null || defaultValue.isNull()) {
+      return null;
+    }
+
+    switch (type.typeId()) {
+      case BOOLEAN:
+        Preconditions.checkArgument(defaultValue.isBoolean(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.booleanValue();
+      case INTEGER:
+        Preconditions.checkArgument(defaultValue.isIntegralNumber() && defaultValue.canConvertToInt(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.intValue();
+      case LONG:
+        Preconditions.checkArgument(defaultValue.isIntegralNumber() && defaultValue.canConvertToLong(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.longValue();
+      case FLOAT:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.floatValue();
+      case DOUBLE:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.doubleValue();
+      case DECIMAL:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.decimalValue();
+      case STRING:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.textValue();
+      case UUID:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return UUID.fromString(defaultValue.textValue());
+      case DATE:
+      case TIME:
+      case TIMESTAMP:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return Literal.of(defaultValue.textValue()).to(type).value();
```

Comment:
Rather than using `Literal`, could you just refactor to add the conversions to `DateTimeUtil` like the to string conversions? That way we have both in the util.